src/target/riscv/riscv.c

```diff
@@ -2916,6 +2915,31 @@ COMMAND_HANDLER(riscv_use_bscan_tunnel)
 	return ERROR_OK;
 }
 
+COMMAND_HANDLER(riscv_mask_interrupts)
+{
+	struct target *target = get_current_target(CMD_CTX);
+	RISCV_INFO(info);
+
+	static const struct jim_nvp nvp_maskisr_modes[] = {
+		{ .name = "off", .value = RISCV_ISRMASK_OFF },
+		{ .name = "steponly", .value = RISCV_ISRMASK_STEPONLY },
+		{ .name = NULL, .value = -1 },
+	};
+	const struct jim_nvp *n;
+
+	if (CMD_ARGC > 0) {
+		n = jim_nvp_name2value_simple(nvp_maskisr_modes, CMD_ARGV[0]);
+		if (!n->name)
+			return ERROR_COMMAND_SYNTAX_ERROR;
+		info->isrmask_mode = n->value;
+	}
+
+	n = jim_nvp_value2name_simple(nvp_maskisr_modes, info->isrmask_mode);
+	command_print(CMD, "riscv interrupt mask %s", n->name);
```

Comment:
Is that expected? I don't think any of the riscv commands echo their result. Should they?

Reply:
Printing result allows us to get the current value without setting.  It can be useful for the tests like `store_existing_setting, change_it, restore_previous_value` 

`riscv maskisr` --> will return existing value

Also I think it is good to see the result in the telnet command line.

Reply:
Another option for your consideration: You may display the current setting if the command is called without any parameters. For example commands OpenOCD's commands `tcl_port`, `telnet_port` (and some others) work this way. Unfortunately, OpenOCD itself is not consistent in this. In any case, it would be good to handle this consistently at least within the riscv target.

Reply:
Ok. changed the command as `set_maskisr` and did not print information for set.